### PR TITLE
Fix(backend,remix): Issues with v5 and resources loading in request object

### PR DESCRIPTION
**Pull request by dimkl**
## Description

As part of this PR the following changes were applied:
- Dropped `user`, `session`, `organization` resources from the SignedOut authObject
- Fix setting the `user`, `session`, `organization` resources in remix `remix`

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [x] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [x] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: .changeset/afraid-experts-eat.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    export const getServerData: GetServerData<any> = withServerAuth(
+        async props => {
+            - const { user, organization, session } =  props;
+            + const { userId, organizationId, sessionId } = props;
+            return { props: { data: '1', auth: props.auth, userId, organizationId, sessionId } };
+        },
+        { loadUser: true },
+    );
+```
```
Comment (dimkl) [inline .changeset/afraid-experts-eat.md:51]: cc: @LekoArts Could you handle adding these changes in clerk-docs?

File: .changeset/strange-trains-bow.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@clerk/remix': patch
+---
+
+Support using env variables in `getAuth` and fix setting resources (`user`/`sessions`/organization`) into `request`
```
Comment (dimkl) [inline .changeset/strange-trains-bow.md:5]: cc: @LekoArts Could you handle adding this change in clerk-docs?

File: .changeset/strange-trains-bow.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@clerk/remix': patch
+---
+
+Support using env variables in `getAuth` and fix setting resources (`user`/`sessions`/`organization`) into `request`.
```
Comment (dimkl) [inline .changeset/strange-trains-bow.md:5]: cc: @LekoArts Could you handle adding these changes in clerk-docs?

Comment (LekoArts) [reply .changeset/strange-trains-bow.md:5]: What do you mean by "support using env variables in `getAuth`"?

File: .changeset/afraid-experts-eat.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    // or 
+    rootAuthLoader(
+        args,
+        ({ request }) => {
+            - const { user, organization, session } = request.auth;
+            + const { userId, organizationId, sessionId } = request.auth;
+            // ...
+        },
+        { loadUser: true },
+    );
```
[the remark this answers is not in the document]
Comment (dimkl) [reply .changeset/afraid-experts-eat.md:32]: The `request.user` is populated when `loadUser: true` is passed. The above example is to showcase that we have dropped the `{ user, organization, session }` from `request.auth`. This removal was applied to make the `SignedInAuthObject` and `SignedOutAuthObject` have the same keys.

File: .changeset/strange-trains-bow.md (hunk 3 above)
Comment (dimkl) [reply .changeset/strange-trains-bow.md:5]: Ignore this. I will replace it with:
> Fix adding `user`/`sessions`/`organization`  resources into `request`

The `support using env variables in getAuth` part was already working. 
PS With that phrase, i meant that i passed the env variables (eg CLERK_SECRET_KEY) as options in the `authenticateRequest` called from `getAuth`.

**Review by panteliselef — APPROVED** [review]

File: .changeset/afraid-experts-eat.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    // NextJS
+    import { getAuth } from '@clerk/nextjs/server';
+    
+    - const { user, organization, session } = getAuth(args);
+    + const { userId, organizationId, sessionId } = getAuth(req, opts);
```
Comment (nikosdouvlis) [inline .changeset/afraid-experts-eat.md:38]: Just checking, did we have `getAuth.user` before v5? 
Nitpick, but this should probably be on its own changeset targeting nextjs (similar for remix and gatsby)

**Review by nikosdouvlis — APPROVED** [review]

File: .changeset/afraid-experts-eat.md (hunk 5 above)
Comment (dimkl) [reply .changeset/afraid-experts-eat.md:38]: Yes we had them in v4.
signed-in: https://github.com/clerk/javascript/blob/c59a2d4a22166076340b3c5c5e20c112a138acbe/packages/backend/src/tokens/authObjects.ts#L120
signed-out: https://github.com/clerk/javascript/blob/c59a2d4a22166076340b3c5c5e20c112a138acbe/packages/backend/src/tokens/authObjects.ts#L142

The change is applied only in the backend package and the changeset will add this change in the changelog of all the packages depending on backend, that's why i didn't add both remix, nextjs, gatsby.
@nikosdouvlis Do we want for each breaking change in backend to also add the other packages?
